Sm oct102020

SMoorthi-emc opened the pull request:
This draft PR updates 
" GFS_surface_composites.F90" and " GFS_surface_composites.meta" to handle different min_seaice and min_lake ice for both coupled an uncoupled and fractional and non-fractional grid.   It also includes Shan';s update to slmsk, after update to islmsk in this code.
It takes care of a potential error Dom pointed out with respect islmsk_cice that is entering the "sfc_sice.f".

climbfuji commented on physics/GFS_surface_composites.F90:
```diff
@@ -79,45 +79,70 @@ subroutine GFS_surface_composites_pre_run (im, lkm, frac_grid, flag_cice, cplflx
             endif
           else
             cice(i) = zero
+            icy(i)  = .false.
```
I am not sure these `icy(i)=.false.` and `wet(i)=.false.` are necessary, that should be the default values.

climbfuji commented on physics/GFS_surface_composites.F90:
```diff
@@ -79,45 +79,70 @@ subroutine GFS_surface_composites_pre_run (im, lkm, frac_grid, flag_cice, cplflx
             dry(i)    = .true.
             frland(i) = one
             cice(i)   = zero
+            icy(i)    = .false.
```
I am not sure these `icy(i)=.false.` and `wet(i)=.false.` are necessary, that should be the default values.

climbfuji commented on physics/GFS_surface_composites.F90:
```diff
@@ -127,19 +152,45 @@ subroutine GFS_surface_composites_pre_run (im, lkm, frac_grid, flag_cice, cplflx
                 flag_cice(i)   = .false.
                 islmsk(i)      = 0
+                islmsk_cice(i) = 0
+                icy(i)         = .false.
```
I am not sure this `icy(i)=.false.` is necessary, that should be the default value.

climbfuji commented on physics/GFS_surface_composites.F90:
```diff
@@ -127,19 +152,45 @@ subroutine GFS_surface_composites_pre_run (im, lkm, frac_grid, flag_cice, cplflx
+              if (oceanfrac(i) > zero .and. .not. cplflx) then
+                if (cice(i) > min_seaice) then
+                  icy(i) = .true.
+                  wet(i) = .false.
```
I am not sure this `wet(i)=.false.` is necessary, that should be the default value.

climbfuji commented on physics/GFS_surface_composites.F90:
```diff
@@ -127,19 +152,45 @@ subroutine GFS_surface_composites_pre_run (im, lkm, frac_grid, flag_cice, cplflx
+                  islmsk_cice(i) = 2
+                else
+                  cice(i)   = zero
+                  icy(i)    = .false.
```
I am not sure this `icy(i)=.false.` is necessary, that should be the default value.

climbfuji commented on physics/GFS_surface_composites.F90:
```diff
@@ -127,19 +152,45 @@ subroutine GFS_surface_composites_pre_run (im, lkm, frac_grid, flag_cice, cplflx
+                endif
+              elseif (cice(i) >= min_lakeice) then
                 icy(i) = .true.
+                wet(i) = .false.
```
I am not sure this `wet(i)=.false.` is necessary, that should be the default value.

climbfuji commented on physics/GFS_surface_composites.F90:
```diff
@@ -127,19 +152,45 @@ subroutine GFS_surface_composites_pre_run (im, lkm, frac_grid, flag_cice, cplflx
               else
                 cice(i)   = zero
-                islmsk(i) = 0
+                icy(i)    = .false.
```
I am not sure this `icy(i)=.false.` is necessary, that should be the default value.

climbfuji commented on physics/micro_mg3_0.F90_Sep19:
```diff
@@ -0,0 +1,4529 @@
+!>\file micro_mg3_0.F90
```
This file `physics/micro_mg3_0.F90_Sep19` will need to be deleted before we merge the PR.

climbfuji reviewed (commented):
A first pass through the changes. I am not sure any of the `icy/wet=.false.` are required, because these should be the default values, but maybe I missed something.

I want to note that there are some unrelated small changes to MG3. It's fine to include them, but we should not that later in the PR.

I can formally combine @shansun6 's and your PRs into one and test if removing the `.false.` statements has any impact on the results.

SMoorthi-emc commented:
Dom,  you are probably right about the default values for ice/wet, but I
wanted to be sure.
Yes, I forgot about MG3; that was to speed up MG3 (It was already in my
latest branch) by not doing calculation where not needed.
Moorthi

On Mon, Oct 19, 2020 at 11:09 AM Dom Heinzeller <notifications@github.com>
wrote:

> *@climbfuji* commented on this pull request.
>
> A first pass through the changes. I am not sure any of the icy/wet=.false.
> are required, because these should be the default values, but maybe I
> missed something.
>
> I want to note that there are some unrelated small changes to MG3. It's
> fine to include them, but we should not that later in the PR.
>
> I can formally combine @shansun6 <https://github.com/shansun6> 's and
> your PRs into one and test if removing the .false. statements has any
> impact on the results.
> ------------------------------
>
> In physics/GFS_surface_composites.F90
> <https://github.com/NCAR/ccpp-physics/pull/512#discussion_r507821922>:
>
> >                else
>                  cice(i)        = zero
>                  flag_cice(i)   = .false.
> -!               islmsk_cice(i) = 0
> -!               islmsk(i)      = 0
> -                wet(i) = .true. ! some open ocean/lake water exists
> +                islmsk_cice(i) = 0
>
> This is good, I noticed this difference to the IPD develop code as well.
> ------------------------------
>
> In physics/GFS_surface_composites.F90
> <https://github.com/NCAR/ccpp-physics/pull/512#discussion_r507823485>:
>
> >                endif
>              else
> -              if (cice(i) >= min_lakeice) then
> +              if (oceanfrac(i) > zero .and. .not. cplflx) then
> +                if (cice(i) >= min_seaice) then
> +                  icy(i) = .true.
> +                  if (cice(i) < one) then
> +                    wet(i) = .true. ! some open ocean exists
> +                    tsfco(i) = max(tsfco(i), tisfc(i), tgice)
> +                  endif
> +                  islmsk_cice(i) = 2
> +                  islmsk(i)      = 2
> +                else
> +                  cice(i)   = zero
> +                  icy(i)  = .false.
>
> I am not sure this icy(i)=.false. is necessary, that should be the
> default value.
> ------------------------------
>
> In physics/GFS_surface_composites.F90
> <https://github.com/NCAR/ccpp-physics/pull/512#discussion_r507823600>:
>
> >                else
>                  cice(i)   = zero
> -!               islmsk(i) = 0
> -                wet(i)    = .true. ! some open ocean/lake water exists
> -              endif
> -            endif
> -            if (wet(i) .and. .not. cplflx) then
> -              if (oceanfrac(i) > zero) then
> -                tsfco(i) = max(tsfco(i), tisfc(i), tgice)
> -              elseif (icy(i)) then
> -                 tsfco(i) = max(tisfc(i), tgice)
> +                icy(i)    = .false.
>
> I am not sure this icy(i)=.false. is necessary, that should be the
> default value.
> ------------------------------
>
> In physics/GFS_surface_composites.F90
> <https://github.com/NCAR/ccpp-physics/pull/512#discussion_r507825613>:
>
> >                endif
>              endif
>            else
>              cice(i) = zero
> +            icy(i)  = .false.
>
> I am not sure these icy(i)=.false. and wet(i)=.false. are necessary, that
> should be the default values.
> ------------------------------
>
> In physics/GFS_surface_composites.F90
> <https://github.com/NCAR/ccpp-physics/pull/512#discussion_r507826126>:
>
> >              dry(i)    = .true.
>              frland(i) = one
>              cice(i)   = zero
> +            icy(i)    = .false.
>
> I am not sure these icy(i)=.false. and wet(i)=.false. are necessary, that
> should be the default values.
> ------------------------------
>
> In physics/GFS_surface_composites.F90
> <https://github.com/NCAR/ccpp-physics/pull/512#discussion_r507826536>:
>
> > @@ -127,19 +152,45 @@ subroutine GFS_surface_composites_pre_run (im, lkm, frac_grid, flag_cice, cplflx
>                  cice(i)        = zero
>                  flag_cice(i)   = .false.
>                  islmsk(i)      = 0
> +                islmsk_cice(i) = 0
> +                icy(i)         = .false.
>
> I am not sure this icy(i)=.false. is necessary, that should be the
> default value.
> ------------------------------
>
> In physics/GFS_surface_composites.F90
> <https://github.com/NCAR/ccpp-physics/pull/512#discussion_r507826693>:
>
> >                endif
>              else
> -              if (cice(i) > min_lakeice) then
> +              if (oceanfrac(i) > zero .and. .not. cplflx) then
> +                if (cice(i) > min_seaice) then
> +                  icy(i) = .true.
> +                  wet(i) = .false.
>
> I am not sure this wet(i)=.false. is necessary, that should be the
> default value.
> ------------------------------
>
> In physics/GFS_surface_composites.F90
> <https://github.com/NCAR/ccpp-physics/pull/512#discussion_r507826894>:
>
> >                endif
>              else
> -              if (cice(i) > min_lakeice) then
> +              if (oceanfrac(i) > zero .and. .not. cplflx) then
> +                if (cice(i) > min_seaice) then
> +                  icy(i) = .true.
> +                  wet(i) = .false.
> +                  if (cice(i) < one) then
> +                    wet(i) = .true. ! some open ocean exists
> +                    tsfco(i) = max(tisfc(i), tgice)
> +                  endif
> +                  islmsk(i)      = 2
> +                  islmsk_cice(i) = 2
> +                else
> +                  cice(i)   = zero
> +                  icy(i)    = .false.
>
> I am not sure this icy(i)=.false. is necessary, that should be the
> default value.
> ------------------------------
>
> In physics/GFS_surface_composites.F90
> <https://github.com/NCAR/ccpp-physics/pull/512#discussion_r507827033>:
>
> >                  icy(i) = .true.
> +                wet(i) = .false.
>
> I am not sure this wet(i)=.false. is necessary, that should be the
> default value.
> ------------------------------
>
> In physics/GFS_surface_composites.F90
> <https://github.com/NCAR/ccpp-physics/pull/512#discussion_r507827142>:
>
> >                else
>                  cice(i)   = zero
> -                islmsk(i) = 0
> +                icy(i)    = .false.
>
> I am not sure this icy(i)=.false. is necessary, that should be the
> default value.
> ------------------------------
>
> In physics/micro_mg3_0.F90_Sep19
> <https://github.com/NCAR/ccpp-physics/pull/512#discussion_r507828591>:
>
> > @@ -0,0 +1,4529 @@
> +!>\file micro_mg3_0.F90
>
> This file physics/micro_mg3_0.F90_Sep19 will need to be deleted before we
> merge the PR.
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/NCAR/ccpp-physics/pull/512#pullrequestreview-511857154>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALLVRYTEIZZ5LGCEPX2DASTSLRJBZANCNFSM4SWIBLBQ>
> .
>


-- 
Dr. Shrinivas Moorthi
Research Meteorologist
Modeling and Data Assimilation Branch
Environmental Modeling Center / National Centers for Environmental
Prediction
5830 University Research Court - (W/NP23), College Park MD 20740 USA
Tel: (301)683-3718

e-mail: Shrinivas.Moorthi@noaa.gov
Phone:  (301) 683-3718   Fax: (301) 683-3718

climbfuji commented:
Is this PR still relevant, or do the changes in https://github.com/NCAR/ccpp-physics/pull/520 cover everything that is needed? The one-liner `tsfc` change from this PR was merged previously as part of the HWRF physics update (https://github.com/NCAR/ccpp-physics/pull/509)

SMoorthi-emc commented:
Dom,
   There may still be something in there that is relevant, but this
particular branch is not relevant anymore as I have updated to a new branch
which is up to date with ccpp-physics master as of yesterday.  Any updates
will come in the future after I have updated to the new ufs-weather-model.
Moorthi

On Fri, Nov 20, 2020 at 9:28 AM Dom Heinzeller <notifications@github.com>
wrote:

> Is this PR still relevant, or do the changes in #520
> <https://github.com/NCAR/ccpp-physics/pull/520> cover everything that is
> needed? The one-liner tsfc change from this PR was merged previously as
> part of the HWRF physics update (#509
> <https://github.com/NCAR/ccpp-physics/pull/509>)
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly, view it on GitHub
> <https://github.com/NCAR/ccpp-physics/pull/512#issuecomment-731201846>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALLVRYSFYNP4E4TSS5ILEDTSQZ4HFANCNFSM4SWIBLBQ>
> .
>


-- 
Dr. Shrinivas Moorthi
Research Meteorologist
Modeling and Data Assimilation Branch
Environmental Modeling Center / National Centers for Environmental
Prediction
5830 University Research Court - (W/NP23), College Park MD 20740 USA
Tel: (301)683-3718

e-mail: Shrinivas.Moorthi@noaa.gov
Phone:  (301) 683-3718   Fax: (301) 683-3718

climbfuji commented:
Thanks, @SMoorthi-emc - it's good to keep the house tidy and close out old PRs. I'll add you as reviewer to @shansun6 's PR, I do have some concerns about a few changes in that one.

shansun6 commented:
Hi Moorthi,

Let me know if you have a new version of GFS_surface_composites.F90 that
you want me to merge into.

Thanks,
Shan

On Fri, Nov 20, 2020 at 7:41 AM Dom Heinzeller <notifications@github.com>
wrote:

> Thanks, @SMoorthi-emc <https://github.com/SMoorthi-emc> - it's good to
> keep the house tidy and close out old PRs. I'll add you as reviewer to
> @shansun6 <https://github.com/shansun6> 's PR, I do have some concerns
> about a few changes in that one.
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/NCAR/ccpp-physics/pull/512#issuecomment-731209189>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALORMVVLMCZRKOMGOB3AZTLSQZ5Y5ANCNFSM4SWIBLBQ>
> .
>

SMoorthi-emc commented:
Shan,  No,  I don't.
Thanks
Moorthi

On Fri, Nov 20, 2020 at 2:43 PM shansun6 <notifications@github.com> wrote:

> Hi Moorthi,
>
> Let me know if you have a new version of GFS_surface_composites.F90 that
> you want me to merge into.
>
> Thanks,
> Shan
>
> On Fri, Nov 20, 2020 at 7:41 AM Dom Heinzeller <notifications@github.com>
> wrote:
>
> > Thanks, @SMoorthi-emc <https://github.com/SMoorthi-emc> - it's good to
> > keep the house tidy and close out old PRs. I'll add you as reviewer to
> > @shansun6 <https://github.com/shansun6> 's PR, I do have some concerns
> > about a few changes in that one.
> >
> > —
> > You are receiving this because you were mentioned.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/NCAR/ccpp-physics/pull/512#issuecomment-731209189>,
> > or unsubscribe
> > <
> https://github.com/notifications/unsubscribe-auth/ALORMVVLMCZRKOMGOB3AZTLSQZ5Y5ANCNFSM4SWIBLBQ
> >
> > .
> >
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/NCAR/ccpp-physics/pull/512#issuecomment-731371716>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALLVRYSGSPBNP25N6ZYCQRTSQ3BHPANCNFSM4SWIBLBQ>
> .
>


-- 
Dr. Shrinivas Moorthi
Research Meteorologist
Modeling and Data Assimilation Branch
Environmental Modeling Center / National Centers for Environmental
Prediction
5830 University Research Court - (W/NP23), College Park MD 20740 USA
Tel: (301)683-3718

e-mail: Shrinivas.Moorthi@noaa.gov
Phone:  (301) 683-3718   Fax: (301) 683-3718

shansun6 commented:
Hi Dom and Moorthi,

Thanks for your comments.

Would it be acceptable to both of you, if I revert (i.e., remove) "!" from
lines 190-224, total 7 of them, in this commit, see below?

    189         if (wet(i)) then                   ! Water
    190 !         uustar_wat(i) = uustar(i)
    191             zorl_wat(i) = zorlo(i)
    192             tsfc_wat(i) = tsfco(i)
    193            tsurf_wat(i) = tsfco(i)
    194 !          weasd_wat(i) = weasd(i)
    195 !          snowd_wat(i) = snowd(i)
    196            weasd_wat(i) = zero
    197            snowd_wat(i) = zero
    198            semis_wat(i) = 0.984_kind_phys
    199 !            qss_wat(i) = qss(i)
    200 !           hflx_wat(i) = hflx(i)
    201         endif
    202         if (dry(i)) then                   ! Land
    203           uustar_lnd(i) = uustar(i)
    204            weasd_lnd(i) = weasd(i)
    205             zorl_lnd(i) = zorll(i)
    206             tsfc_lnd(i) = tsfcl(i)
    207            tsurf_lnd(i) = tsfcl(i)
    208            snowd_lnd(i) = snowd(i)
    209            semis_lnd(i) = semis_rad(i)
    210 !            qss_lnd(i) = qss(i)
    211 !           hflx_lnd(i) = hflx(i)
    212         end if

Thanks,
Shan

On Fri, Nov 20, 2020 at 12:45 PM SMoorthi-emc <notifications@github.com>
wrote:

> Shan, No, I don't.
> Thanks
> Moorthi
>
> On Fri, Nov 20, 2020 at 2:43 PM shansun6 <notifications@github.com> wrote:
>
> > Hi Moorthi,
> >
> > Let me know if you have a new version of GFS_surface_composites.F90 that
> > you want me to merge into.
> >
> > Thanks,
> > Shan
> >
> > On Fri, Nov 20, 2020 at 7:41 AM Dom Heinzeller <notifications@github.com
> >
> > wrote:
> >
> > > Thanks, @SMoorthi-emc <https://github.com/SMoorthi-emc> - it's good to
> > > keep the house tidy and close out old PRs. I'll add you as reviewer to
> > > @shansun6 <https://github.com/shansun6> 's PR, I do have some concerns
> > > about a few changes in that one.
> > >
> > > —
> > > You are receiving this because you were mentioned.
> > > Reply to this email directly, view it on GitHub
> > > <https://github.com/NCAR/ccpp-physics/pull/512#issuecomment-731209189
> >,
> > > or unsubscribe
> > > <
> >
> https://github.com/notifications/unsubscribe-auth/ALORMVVLMCZRKOMGOB3AZTLSQZ5Y5ANCNFSM4SWIBLBQ
> > >
> > > .
> > >
> >
> > —
> > You are receiving this because you were mentioned.
> > Reply to this email directly, view it on GitHub
> > <https://github.com/NCAR/ccpp-physics/pull/512#issuecomment-731371716>,
> > or unsubscribe
> > <
> https://github.com/notifications/unsubscribe-auth/ALLVRYSGSPBNP25N6ZYCQRTSQ3BHPANCNFSM4SWIBLBQ
> >
> > .
> >
>
>
> --
> Dr. Shrinivas Moorthi
> Research Meteorologist
> Modeling and Data Assimilation Branch
> Environmental Modeling Center / National Centers for Environmental
> Prediction
> 5830 University Research Court - (W/NP23), College Park MD 20740 USA
> Tel: (301)683-3718
>
> e-mail: Shrinivas.Moorthi@noaa.gov
> Phone: (301) 683-3718 Fax: (301) 683-3718
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly, view it on GitHub
> <https://github.com/NCAR/ccpp-physics/pull/512#issuecomment-731372366>,
> or unsubscribe
> <https://github.com/notifications/unsubscribe-auth/ALORMVW6SWUI4PEJCCFXQMDSQ3BNDANCNFSM4SWIBLBQ>
> .
>

climbfuji commented:
> Hi Dom and Moorthi, Thanks for your comments. Would it be acceptable to both of you, if I revert (i.e., remove) "!" from lines 190-224, total 7 of them, in this commit, see below? 189 if (wet(i)) then ! Water 190 ! uustar_wat(i) = uustar(i) 191 zorl_wat(i) = zorlo(i) 192 tsfc_wat(i) = tsfco(i) 193 tsurf_wat(i) = tsfco(i) 194 ! weasd_wat(i) = weasd(i) 195 ! snowd_wat(i) = snowd(i) 196 weasd_wat(i) = zero 197 snowd_wat(i) = zero 198 semis_wat(i) = 0.984_kind_phys 199 ! qss_wat(i) = qss(i) 200 ! hflx_wat(i) = hflx(i) 201 endif 202 if (dry(i)) then ! Land 203 uustar_lnd(i) = uustar(i) 204 weasd_lnd(i) = weasd(i) 205 zorl_lnd(i) = zorll(i) 206 tsfc_lnd(i) = tsfcl(i) 207 tsurf_lnd(i) = tsfcl(i) 208 snowd_lnd(i) = snowd(i) 209 semis_lnd(i) = semis_rad(i) 210 ! qss_lnd(i) = qss(i) 211 ! hflx_lnd(i) = hflx(i) 212 end if Thanks, Shan On Fri, Nov 20, 2020 at 12:45 PM SMoorthi-emc <notifications@github.com> wrote:
> […](#)

I am going to answer in PR https://github.com/NCAR/ccpp-physics/pull/520, ok?

SMoorthi-emc commented:
Can someone tell how to compile the ufs-weather-model in uncoupled and coupled model on wcoss dell?
I can't find any instruction anywhere and it is extremely frustrating.

climbfuji commented:
> Can someone tell how to compile the ufs-weather-model in uncoupled and coupled model on wcoss dell?
> I can't find any instruction anywhere and it is extremely frustrating.

I am not sure this is the right place to ask this question (ccpp-physics). The ufs-weather-model GitHub repo might be a better place to get help, in particular because most people here don't have access to WCOSS. My guess is to go to subdirectory `tests` and try:
```
./compile.sh wcoss_dell_p3 "CCPP=Y SUITES=..." [""] [NO] [NO] 2>&1 | tee compile.log
```
Arguments in square brackets are optiona. The empty `""` are the label for the executable and the `modules.fv3` file, `NO NO` means no clean before and no clean after. By default the label is empty and `clean_before` and `clean_after` are `YES`. If you don't specify `SUITES=...` it will compile all suites. And you know the optional `DEBUG=Y` or `REPRO=Y` that can go into the `MAKEOPT` string.

If you want to know how to compile in coupled mode, look inside `rt.conf` for the `COMPILE` line that corresponds to what you want to do. For Cray, replace `wcoss_dell_p3` with `wcoss_cray`. As I said, my best guess because I don't have access to WCOSS. Hope this helps.
